File: 049_group_anagrams/sort.py
class Solution:
    def groupAnagrams(self, strs):
        """
        :type strs: List[str]
        :rtype: List[List[str]]
        """
        """
        Complexity: O(nklgk), n: length of strs, k: max length of str
        Space: O(nk)
        """
        ans = collections.defaultdict(list)
        for str in strs:
            ans[tuple(sorted(str))].append(str)
        return list(ans.values())

File: 049_group_anagrams/sort.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string, vector<string> > map;
        for(auto str : strs){
            string cpy = str;
            sort(str.begin(), str.end());
            map[str].push_back(cpy);
        }
        vector< vector<string> > ans;
        for(auto it : map){
            ans.push_back(it.second);
        }
        return ans;
    }
};
